src/domain/targeted-messaging/targeted-messaging.repository.ts

```diff
@@ -0,0 +1,48 @@
+import { ITargetedMessagingDatasource } from '@/domain/interfaces/targeted-messaging.datasource.interface';
+import { ISafeRepository } from '@/domain/safe/safe.repository.interface';
+import { Submission } from '@/domain/targeted-messaging/entities/submission.entity';
+import { TargetedSafe } from '@/domain/targeted-messaging/entities/targeted-safe.entity';
+import { ITargetedMessagingRepository } from '@/domain/targeted-messaging/targeted-messaging.repository.interface';
+import { BadRequestException, Inject, Injectable } from '@nestjs/common';
+
+@Injectable()
+export class TargetedMessagingRepository
+  implements ITargetedMessagingRepository
+{
+  constructor(
+    @Inject(ITargetedMessagingDatasource)
+    private readonly datasource: ITargetedMessagingDatasource,
+    @Inject(ISafeRepository)
+    private readonly safeRepository: ISafeRepository,
+  ) {}
+
+  async getTargetedSafe(args: {
+    outreachId: number;
+    safeAddress: `0x${string}`;
+  }): Promise<TargetedSafe> {
+    return this.datasource.getTargetedSafe(args);
+  }
+
+  async getSubmission(args: {
+    chainId: string;
+    targetedSafe: TargetedSafe;
+    signerAddress: `0x${string}`;
+  }): Promise<Submission> {
+    const isOwner = await this.safeRepository.isOwner({
+      chainId: args.chainId,
+      safeAddress: args.targetedSafe.address,
+      address: args.signerAddress,
+    });
+    if (!isOwner) {
+      throw new BadRequestException('The signer is not a Safe owner');
+    }
+    return this.datasource.getSubmission(args);
+  }
+
+  async createSubmission(args: {
+    targetedSafe: TargetedSafe;
+    signerAddress: `0x${string}`;
+  }): Promise<Submission> {
+    return this.datasource.createSubmission(args);
```

Comment:
I think we should also add an owner check here.

Reply:
`getSubmission` is always called before trying to create the submission. There's a test that assures a non-owner cannot create submissions (https://github.com/safe-global/safe-client-gateway/blob/b7acfdbe9a90b04da2795c760caf5d280f13de5c/src/routes/targeted-messaging/targeted-messaging.controller.spec.ts#L395-L396), so if the `getSubmission` code is changed and it stops checking the ownership, the test would fail. 

Happy to change this to explicitly check it on creation though if you prefer, but we would be executing the datasource call twice then. Wdyt?

Reply:
Considering we have the test in place, I think that's sufficient.